src/Entities/DetailedProduct.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+
+namespace RobotStores\Entities;
+
+
+class DetailedProduct extends BaseProduct
+{
+    private ?string $image2;
+
+    private ?string $image3;
+
+    private \RobotStores\Interfaces\CategoryInterface $category;
```

Comment:
put the namespace into a use statement then you wont have to use the fully qualified namespace twice (see getCategory method return type)